trieste/acquisition/function.py

```diff
@@ -600,13 +600,14 @@ def prepare_acquisition_function(
 
         constraint_fn = self._constraint_builder.prepare_acquisition_function(datasets, models)
         pof = constraint_fn(objective_dataset.query_points[:, None, ...])
-        is_feasible = pof >= self._min_feasibility_probability
+        is_feasible = tf.squeeze(pof >= self._min_feasibility_probability, axis=-1)
```

Comment:
Why must this change?

Reply:
tbh i'm not sure. maybe cos we were masking over the target space [N, 1] before and now we're masking over the input space [N] + feature_shape

Reply:
it kinda makes sense anyway, cos `constraint_fn` outputs [..., 1] and we don't need that trailing dim for the mask (indeed my experience has been that `tf.boolean_mask` doesn't like non-vector masks

Reply:
from the [docs](https://www.tensorflow.org/api_docs/python/tf/boolean_mask)

> In general, 0 < dim(mask) = K <= dim(tensor), and mask's shape must match the first K dimensions of tensor's shape.

does that agree with me?

Reply:
you may need to check that, I'm tired and not thinking clearly

Reply:
ah yes, and the mean would have been [N, 1] so there was no need for squeeze